tests/31sync/09archived.pl

```diff
@@ -129,9 +133,10 @@
       matrix_create_filter( $user, {} )->then( sub {
          ( $filter_id ) = @_;
 
-         Future->needs_all(
-            matrix_create_room( $user )->on_done( sub { ( $room_id_1 ) = @_; } ),
-            matrix_create_room( $user )->on_done( sub { ( $room_id_2 ) = @_; } ),
+         matrix_create_room( $user )->on_done( sub { ( $room_id_1 ) = @_; } );
+      })->then( sub {
+         matrix_create_room_and_wait_for_sync($user )->on_done( sub {
+            ( $room_id_2 ) = @_; }
```

Comment:
Any reason these two have to now be serialised, instead of the concurrent form they were in before?